### DISCUSSION TITLE
How to set headers before making a network request on client-side?

### DISCUSSION THREAD

**Discussion by junshengpierre** [trpc/trpc#1402] [Q&A]
Hi, I'm using Next.js and firebase auth for authentication with [next-firebase-auth](https://github.com/gladly-team/next-firebase-auth) and [next-firebase-auth](https://github.com/gladly-team/next-firebase-auth)'s recommended way to make an authenticated request is by passing an auth token in the `authorization` header when making the network request but the auth token is only accessible via a react hook like this:

```javascript
import { withAuthUser } from "next-firebase-auth";

function Page() {
  const AuthUser = useAuthUser();
  
  useEffect(() => {
    async function fetchUser() {
      const authToken = await AuthUser.getIdToken();
      const user = await api.get("/user", {
        headers: {
          Authorization: authToken,
        },
      });
      console.log('user: ', user)
    }
    fetchUser()
  }, []);

  return <div>My Page</div>;
}

export default withAuthUser()(Page);
```

The only way I found to pass headers is to set `authToken` or `AuthUser` in some global storage and then set it in the config of the `withTRPC` but wanted to see if there's an easier way to do it, is there a way to set headers directly before making a network request in `trpc/@react` hooks or trpc client?

**Comment by soulsam480** [comment]
see here, https://github.com/trpc/trpc/discussions/1408, probably that's the answer
